services/ui-src/src/components/fields/NumberField.test.tsx

```diff
@@ -125,6 +129,30 @@ describe("Test Percentage Masked NumberField", () => {
   });
 });
 
+describe("Test Ratio Masked NumberField", () => {
+  test("onChangeHandler updates ratio field value", async () => {
+    const result = render(ratioMaskedNumberFieldComponent);
+    const numberFieldInput: HTMLInputElement = result.container.querySelector(
+      "[name='testNumberField']"
+    )!;
+    await userEvent.type(numberFieldInput, "123:123");
+    expect(numberFieldInput.value).toEqual("123:123");
+    await userEvent.tab();
+    expect(numberFieldInput.value).toEqual("123:123");
+    await userEvent.clear(numberFieldInput);
+    await userEvent.type(
+      numberFieldInput,
+      "123,,,4567.1234567.1234:12,3456,7.1"
+    );
+    await userEvent.tab();
+    expect(numberFieldInput.value).toEqual("1,234,567.12:1,234,567.1");
+    await userEvent.clear(numberFieldInput);
+    await userEvent.type(numberFieldInput, ":");
+    await userEvent.tab();
+    expect(numberFieldInput.value).toEqual(":");
```

Comment:
Do we want this to be the case? A single colon is listed in the "bad test cases" below

Reply:
I guess maybe it's ok to leave it, it'll just fail validation, right?

Reply:
Right, that'll fail validation so it wouldn't get through on the form. Basically this test is to make sure the masking gets called and that mask returns the right value back when you tab away.